src/ocean/sys/stats/linux/ProcVFS.d

```diff
@@ -78,9 +80,38 @@ public int getOpenFdCount ()
 {
     int count;
 
-    foreach (c; Path.children("/proc/self/fdinfo"))
+    auto dir = opendir("/proc/self/fdinfo".ptr);
+
+    if (dir is null)
+        return 0;
+
+    scope (exit)
     {
-        count++;
+        // ignore EBADF
+        closedir(dir);
+    }
+
+    /*
+        From http://man7.org/linux/man-pages/man3/readdir_r.3.html
+
+        In the current POSIX.1 specification (POSIX.1-2008), readdir(3) is
+        not required to be thread-safe. However, in modern
+        implementations (including the glibc implementation), concurrent
+        calls to readdir(3) that specify different directory streams are
+        thread-safe. Therefore, the use of readdir_r() is generally
+        unnecessary in multithreaded programs. In cases where multiple
+        threads must read from the same directory stream, using readdir(3)
+        with external synchronization is still preferable to the use of
+        readdir_r(), for the reasons given in the points above.
+    */
+    dirent* dp;
+    while ((dp = readdir(dir)) !is null)
+    {
+        if (strcmp(dp.d_name.ptr, "..".ptr) != 0 &&
```

Comment:
Tiny point: I'm wondering if `StringC.toDString(dp.d_name.ptr) == ".."` would be more idiomatic and slightly easier to read. (Or was there a more modern / favoured alternative to `ocean.text.util.StringC`?)

Reply:
Fair enough, I'll replace it with `toDString`.